DOC: add note for torch.clamp() special case min > max See #45664

xamm opened the pull request:
Fixes #45664

This PR adds a note to the documentation for `torch.clamp()` to alert users to a special case: If `min` is greater than `max`, all values are set to the `max` value.

Also, an example was added after the first code example. And this one is referenced in the note.

xamm committed:
add note for torch.clamp() special case min > max See #45664

xamm commented on torch/_torch_docs.py:
```diff
@@ -1916,6 +1922,8 @@ def merge_dicts(*dicts):
     tensor([-1.7120,  0.1734, -0.0478, -0.0922])
     >>> torch.clamp(a, min=-0.5, max=0.5)
     tensor([-0.5000,  0.1734, -0.0478, -0.0922])
+    >>> torch.clamp(a, min=8.0, max=11.0)
```
Should there be an example for this special case or is the note sufficient?

mruberry replied:
I think a note alone makes sense because this is a degenerate case we don't want to highlight with an example

xamm replied:
I removed this with the last commit.

mruberry commented on torch/_torch_docs.py:
```diff
@@ -1901,6 +1901,12 @@ def merge_dicts(*dicts):
 
+.. note::
+    If :attr:`min` is greater than :attr:`max` :func:`torch.clamp(..., min, max) <torch.clamp>`
+    sets all values equal to :attr:`max`.
```
"all values" -> "all output values"?

xamm replied:
Maybe "sets all elements in `input` to `max`." Would be better to be more consistent with the general description that `torch.clamp()` 

> Clamp all elements in input into the range...

mruberry commented on torch/_torch_docs.py:
```diff
@@ -1901,6 +1901,12 @@ def merge_dicts(*dicts):
+    If :attr:`min` is greater than :attr:`max` :func:`torch.clamp(..., min, max) <torch.clamp>`
+    sets all values equal to :attr:`max`.
+
+    This is shown in the second example below.
```
We can probably get rid of this sentence

xamm replied:
The reference to the example is deleted with the example itself.

mruberry reviewed (commented):
Thanks for fixing this, @xamm! I made a few small suggestions

xamm committed:
DOC: change clamp description after review

- remove example for special case min > max
- clarify which elements are set to max in this special case

mruberry reviewed (approved):
Neat! Thanks @xamm!

xamm commented:
I was happy to help.
